### PR TITLE
HTML CSS formulaire

### PR DESCRIPTION
**Pull request by khadimb**

### CODE REVIEW

**Review by Erdnaxela45 — APPROVED** [review]
Incroyable

File: public/assets/css/contact.css [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+    }
+}
+
+@media (max-width: 768px) {
+    .subject {
+        width: 70%;
+    }
+}
+
+
```
Comment (LucyMerlier) [inline public/assets/css/contact.css:109]: avoid magic values

File: public/assets/css/contact.css [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+}
+    
+.fas {
+    margin: 25px 10px 0;
+    font-size: 72px;
+    color: #fff;
+}
+    
+input, textarea {
+    width: calc(100% - 18px);
```
Comment (LucyMerlier) [inline public/assets/css/contact.css:48]: why?

File: src/View/Contact/contact.html.twig [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+    <form action="/" class="form" method= "POST">
+        <h1>Contactez moi</h1>
+            <div class="info">
+                <label for="name">Nom complet :</label>
+                <input class="name" type="text" name="name" placeholder="Nom complet" required>
+                <label for="email">Adresse mail :</label>
+                <input class="email" type="text" name="email" placeholder="exemple@exemple.com" required>
+                <label for="tel">Numéro de téléphone :</label>
+                <input class="tel" type="text" name="tel" placeholder="00 00 00 00 00" required>
```
Comment (LucyMerlier) [inline src/View/Contact/contact.html.twig:19]: there is an html type for phone numbers ;)

File: src/Controller/ContactController.php [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+     * @throws \Twig\Error\SyntaxError
+     */
+
+    public function index()
+    {
+
+        if ($_SERVER['REQUEST_METHOD'] === 'POST') {
+            $data = array_map('trim', $_POST);
+            $errors = [];
+            $maxLength = 255;
```
Comment (LucyMerlier) [inline src/Controller/ContactController.php:22]: should be a constant

File: src/View/Contact/contact.html.twig [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+                <label for="subject">Sujet : </label>
+                <select class="subject" name="subject" id="subject">
+                    <option value="">--Veuillez choisir votre sujet--</option>
+                    <option value="tel-session">Séance téléphonique</option>
+                    <option value="cabinet-session">Séance au cabinet</option>
```
Comment (LucyMerlier) [inline src/View/Contact/contact.html.twig:24]: "cabinet-session" => does not sound very british ;)

File: src/Controller/ContactController.php [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+
+            if (empty($data['name'])) {
+                $errors[] = 'Le prénom est obligatoire';
+            } elseif ($data['name'] > $maxLength) {
+                $errors = 'Le prénom doit faire moins de' . $maxLength . ' caractères';
```
Comment (LucyMerlier) [inline src/Controller/ContactController.php:27]: shouldn't there be a space after "de"?

File: src/Controller/ContactController.php [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+            }
+            if (!filter_var($data['email'], FILTER_VALIDATE_EMAIL)) {
+                $errors[] = 'Mauvais format d\'email';
+            }
+            if (empty($data['message'])) {
+                $errors[] = 'Un message est obligatoire';
+            }
+            if (empty($errors)) {
+                echo 'Tout à bien été rempli';
+            }
```
Comment (LucyMerlier) [inline src/Controller/ContactController.php:41]: DON'T ECHO STUFF IN THE CONTROLLER

File: src/Controller/ContactController.php [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+        }
+
+        return $this->twig->render('Contact/contact.html.twig');
```
Comment (LucyMerlier) [inline src/Controller/ContactController.php:44]: what if I use "bibi" as subject?

**Review by LucyMerlier — CHANGES_REQUESTED** [review]
see comments

File: src/View/Contact/contact.html.twig (hunk 5 above)
Comment (khadimb) [reply src/View/Contact/contact.html.twig:24]: this is the right translation

File: src/Controller/ContactController.php (hunk 6 above)
Comment (khadimb) [reply src/Controller/ContactController.php:27]: ok

File: src/Controller/ContactController.php (hunk 7 above)
Comment (khadimb) [reply src/Controller/ContactController.php:41]: ok

File: src/View/Contact/contact.html.twig (hunk 3 above)
Comment (khadimb) [reply src/View/Contact/contact.html.twig:19]: ok

File: public/assets/css/contact.css (hunk 1 above)
Comment (khadimb) [reply public/assets/css/contact.css:109]: ok

File: public/assets/css/contact.css (hunk 2 above)
Comment (khadimb) [reply public/assets/css/contact.css:48]: ok

File: src/Controller/ContactController.php (hunk 8 above)
Comment (khadimb) [reply src/Controller/ContactController.php:44]: ok

File: src/Controller/ContactController.php (hunk 4 above)
Comment (khadimb) [reply src/Controller/ContactController.php:22]: ok

**Pull request closed without merging**